include/swift/AST/DiagnosticsSema.def

```diff
@@ -1739,6 +1739,9 @@ ERROR(swift_native_objc_runtime_base_not_on_root_class,none,
       "@_swift_native_objc_runtime_base_not_on_root_class can only be applied "
       "to root classes", ())
 
+WARNING(objc_implementation_early_spelling_deprecated,none,
+        "warning-only '@_objcImplementation' spelling is deprecated; use "
```

Comment:
This diagnostic is difficult for me to parse, is `warning-only` an intentional inclusion here? I would expect this to just be `'@_objcImplementation' is deprecated; use '@implementation' instead` (I don't think "spelling" is needed given that the diagnostic says "use '@implementation' instead").

Reply:
There's a small functional difference between `@_objcImplementation` and `@implementation`—the former produces warnings, the latter errors—so some early adopters may need to continue using `@_objcImplementation` as a workaround for bugs for a little while longer. I was trying to nod to that in the diagnostic in case they need a reminder of why they can't update yet, but this wording probably doesn't really succeed.

Reply:
Thinking about it more, I think the better thing to do is probably to only emit this warning if there aren't any other diagnostics. That way, early adopters will not get this warning until they no longer have any reason to use `@_objcImplementation`.